internal/service/qbusiness/application.go

```diff
@@ -0,0 +1,422 @@
+				Required:    true,
+			},
+			names.AttrID:                      framework.IDAttribute(),
+			"identity_center_application_arn": framework.ARNAttributeComputedOnly(),
+			"identity_center_instance_arn": schema.StringAttribute{
+				CustomType:  fwtypes.ARNType,
+				Description: "ARN of the IAM Identity Center instance you are either creating for—or connecting to—your Amazon Q Business application",
+				Required:    true,
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+		},
+		Blocks: map[string]schema.Block{
+			"attachments_configuration": schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[attachmentsConfigurationData](ctx),
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.IsRequired(),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"attachments_control_mode": schema.StringAttribute{
+							Required:    true,
+							Description: "Status information about whether file upload functionality is activated or deactivated for your end user.",
+							Validators: []validator.String{
+								enum.FrameworkValidate[awstypes.AttachmentsControlMode](),
+							},
+						},
+					},
+				},
+			},
+			names.AttrEncryptionConfiguration: schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[encryptionConfigurationData](ctx),
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						names.AttrKMSKeyID: schema.StringAttribute{
+							Required:    true,
+							Description: "The identifier of the AWS KMS key that is used to encrypt your data. Amazon Q doesn't support asymmetric keys.",
+							Validators: []validator.String{
+								stringvalidator.LengthBetween(1, 2048),
+							},
+						},
+					},
+				},
+			},
+			names.AttrTimeouts: timeouts.Block(ctx, timeouts.Opts{
+				Create: true,
+				Update: true,
+				Delete: true,
+			}),
+		},
+	}
+}
+
+func (r *resourceApplication) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data resourceApplicationData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	conn := r.Meta().QBusinessClient(ctx)
+
+	input := &qbusiness.CreateApplicationInput{}
+	resp.Diagnostics.Append(fwflex.Expand(ctx, data, input)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	input.Tags = getTagsIn(ctx)
+	input.ClientToken = aws.String(id.UniqueId())
+
+	out, err := conn.CreateApplication(ctx, input)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.QBusiness, create.ErrActionCreating, ResNameApplication, data.DisplayName.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	id := aws.ToString(out.ApplicationId)
```

Comment:
we should set partial state so that the resource taints if waiting fails
```suggestion
	id := aws.ToString(out.ApplicationId)
	resp.State.SetAttribute(ctx, path.Root(names.AttrID), id)
```

Reply:
Can't commit this directly as the `path` import is needed, but will get this fixed.